libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -1002,8 +998,372 @@ bool AP_Frsky_Telem::get_telem_data(uint8_t &frame, uint16_t &appid, uint32_t &d
+    }
+}
+
+/*
+ * Handle the PARAM_SET mavlite message
+ * for FrSky SPort Passthrough (OpenTX) protocol (X-receivers)
+ */
+void AP_Frsky_Telem::mavlite_handle_param_set(mavlite_message_t* rxmsg)
+{
+    char param_name[AP_MAX_NAME_SIZE+1] {};
+    float param_value = 0.0f;
+    // populate packet with mavlite payload
+    mavlite_msg_get_float(rxmsg,&param_value,0);
+    mavlite_msg_get_string(rxmsg,param_name,4);
+    // send message to backend (sync)
+    AP_Param::set_and_save(param_name,param_value);
+    // ok let's read back the last value
+    float value = 0.0f;
+    if (AP_Param::get(param_name,value)) {
+        // initialize
+        mavlite_init_parse(&_mavlite_tx_message,&_mavlite_tx_status);
+        // set msg ID
+        _mavlite_tx_message.msgid=22;
+        // set parameters
+        mavlite_msg_set_float(&_mavlite_tx_message,&value,0);
+        mavlite_msg_set_string(&_mavlite_tx_message,param_name,4);
+        // queue for sending
+        mavlite_send_message(&_mavlite_tx_message,&_mavlite_tx_status);
+    }
+}
+
+/*
+ * Process an incoming mavlite message
+ * for FrSky SPort Passthrough (OpenTX) protocol (X-receivers)
+ */
+void AP_Frsky_Telem::mavlite_process_message(mavlite_message_t* rxmsg)
+{
+    switch (rxmsg->msgid) {
+        case 20: // PARAM_REQUEST_READ
+            mavlite_handle_param_request_read(rxmsg);
+            break;
+        case 23: // PARAM_SET
+            mavlite_handle_param_set(rxmsg);
+            break;
+        case 76: // COMMAND_LONG
+            mavlite_handle_command_long(rxmsg);
+            break;
+    }
+}
+
+/*
+ * Send a mavlite message
+ * Message is chunked in sport packets pushed in the tx queue
+ * for FrSky SPort Passthrough (OpenTX) protocol (X-receivers)
+ */
+bool AP_Frsky_Telem::mavlite_send_message(mavlite_message_t* txmsg, mavlite_status_t* status)
+{
+    // let's check if there's enough room to send it
+    uint32_t space = _sport_tx_packet_buffer.space();
+
+    if (space < SPORT_MAVLITE_MSG_SIZE(txmsg->len)) {
+        return false;
+    }
+    // prevent looping forever
+    uint8_t packet_count = 0;
+    while (status->parse_state != PARSE_STATE_MESSAGE_RECEIVED && packet_count++ < SPORT_MAVLITE_MSG_SIZE(MAVLITE_MAX_PAYLOAD_LEN)) {
+        sport_packet_t packet{};
+        for (uint8_t i=0; i<6; i++) {
+            mavlite_tx_parse(&packet.raw[i+2],i, txmsg, status);
+        }
+        // cache the packet before sending
+        _sport_tx_packet_buffer.push(packet);
+        if (status->parse_state == PARSE_STATE_ERROR) {
+            break;
+        }
+    }
+    status->parse_state = PARSE_STATE_IDLE;
+    return true;
+}
+
+/*
+ * Utility method to apply constraints in changing sensor id values
+ * for FrSky SPort Passthrough (OpenTX) protocol (X-receivers)
+ */
+bool AP_Frsky_Telem::set_sport_sensor_id(AP_Int8 param_idx, uint8_t &sensor)
+{
+    int8_t idx = param_idx.get();
+
+    if (idx == -1) {
```

Comment:
nothing, the algo generates a sensor ID that the system will ignore since it will never receive any polling for it.

First version of this method had all sorts of checks to prevent users from using well known sensor IDs, but later I decided to drop them all for there are many parameters that would crash a vehicle and are not range checked :-)
Anyway no big deal I can enforce some checking.